web/web.go

```diff
@@ -356,7 +354,7 @@ func New(logger *slog.Logger, o *Options) *Handler {
 	factoryAr := func(context.Context) api_v1.AlertmanagerRetriever { return h.notifier }
 	FactoryRr := func(context.Context) api_v1.RulesRetriever { return h.ruleManager }
 
-	var app storage.Appendable
```

Comment:
here is an example where it looks like we could use either one appender or the other -- The perfect place to have a switchable selection for rollbacks.

Reply:
It's a good place IF we want to double down on tech debt of extra ~3k LOC of potentially prod code to keep (most of the code I am removing in this PR would need to be left out). I acknowledge the risk but I propose we don't do dual path anywhere else than scrape and TSDB. We can revert this PR/commit if needed.

Reply:
See: https://github.com/prometheus/prometheus/pull/17675#issuecomment-3668972596